both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
```

Comment:
##### 1

Meteor docs say *"If you define a method on the client with the same name as a server method, Meteor will run it to attempt to predict the outcome of the server method. When the code on the server actually finishes, the prediction generated on the client will be replaced with the actual outcome of the server method."*

Does the **same** method qualify as "a method on the client with the same name as a server method"? The Latency Compensation section is in the Methods chapter, which makes me think it's maybe only for methods.

What would get replaced by what, never mind how, in this case is far from clear to me.

##### 2

Yes, that's the callback I meant.

"Thread" is probably not the right word. Maybe it's "fiber". Terminology aside, what happens is that the client calls the Method, does **not** wait for a response and proceeds to execute the following lines. When the call later returns, the callback (which is both for error and success handling, unlike what I said) is executed in it's own little "world", from which exceptions can't escape.

Reference: https://github.com/meteor/meteor/issues/1640

##### Examples

Most `Meteor.call` calls with an error handler throw an exception. Since the exception has no effect outside the handler, they are at least clumsily written. This common pattern

    if(err) {
      throw err;
    }
    $tr.removeClass('saving');

is better written like this

    if(!err) {
      $tr.removeClass('saving');
    }

I assume these exceptions were written expecting them to propagate out beyond the callback function, but I can't quickly spot an example of where the code relies on that.